app/assets/javascripts/discourse/app/lib/autocomplete.js

```diff
@@ -537,19 +540,20 @@ export default function (options) {
     closeAutocomplete();
   });
 
-  function checkTriggerRule(opts) {
-    return options.triggerRule ? options.triggerRule(me[0], opts) : true;
+  async function checkTriggerRule(opts) {
```

Comment:
Making `checkTriggerRule` async forced all the other `async`/`await` changes in this file.